TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -106,42 +108,43 @@ class MetAPI{
             parameters: [ : ]) { result in
                 switch result {
                 case .failure:
-                    completion(nil)
+                    completion(.failure(.invalidUrlComponents))
                 case .success(let data):
                     let jsonDecoder = JSONDecoder()
                     jsonDecoder.keyDecodingStrategy = .useDefaultKeys
                     do {
                         let result = try jsonDecoder.decode(ObjectResponse.self, from: data)
-                        completion(result)
+                        completion(.success(result))
                     } catch {
-                        completion(nil)
+                        completion(.failure(.noDataInResponse))
                     }
                 }
             }
     }
 
-    func departments(completion: @escaping (DepartmentsResponse?) -> Void) {
-        self.metAPICache.departments { [weak self] departmentsResponce in
+    func departments(completion: @escaping (Result<DepartmentsResponse, MetAPIError>) -> Void) {
+        self.metAPICache.departments { [weak self] departmentsResponse in
             guard let self = self else {
-                completion(nil)
+                completion(.failure(.metAPIDoesNotExist))
                 return
             }
-            if let departmentsResponce = departmentsResponce {
-                completion(departmentsResponce)
+            if let departmentsResponse = departmentsResponse {
+                completion(.success(departmentsResponse))
             } else {
-                self.executeDepartments {[weak self] realResponse in
-                    guard let realResponse = realResponse else {
-                        completion(nil)
-                        return
+                self.executeDepartments {[weak self] realResponseResult in
+                    switch realResponseResult {
+                    case . failure:
+                        completion(.failure(.noDataInResponse))
```

Comment:
This is not a `noDataInResponse`, use the error associated with the `failure` case:
```
case .failure(let error):
```